xamarin/Xamarin.Forms
# [Core] Propagate BindingContext changes to TitleView

### Description of Change ###
If the BindingContext on the Page changes, the new BindingContext isn't propagated to the TitleView. These changes propagate that BindingContext to the TitleView in much the same way that it gets propagated to Menu Items

### Issues Resolved ### 


- fixes #3596 
- fixes #3763

### Platforms Affected ### 


- Core/XAML (all platforms)
- iOS
- Android
- UWP

### Testing Procedure ###
-Run Issue3763 in the control gallery and the instructions will indicate what you should see

### PR Checklist ###
